src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.cs

```diff
@@ -36,13 +36,24 @@ public partial class RegexGenerator : IIncrementalGenerator
             "#pragma warning disable CS0219 // Variable assigned but never used",
         };
 
+        private sealed record CompilationData(bool AllowUnsafe = false, bool CheckOverflow = false);
+
         public void Initialize(IncrementalGeneratorInitializationContext context)
         {
+            // To avoid invalidating every regex's output when anything from the compilation changes,
```

Comment:
@CyrusNajmabadi, can you review this file and make sure I didn't do anything that invalidates your perf efforts?